### PR TITLE
Feature/multiple node groups

### PR DESCRIPTION
**Pull request by shapirov103**
*Issue #, if available:* 236, 242

*Description of changes:*
Mixed node group support
Refactored existing node groups and add-ons that depended on a single node group (Cluster Autoscaler, Container Insights, etc.)
Container insights is leveraging ADOT (open telemetry) and fluentbit
Bug fixes
Doc updates

Branch conflict will be resolved after review, otherwise you will see too many irrelevant commits.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/cluster-providers/generic-cluster-provider.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# Generic Cluster Provider
```
Comment (kcoleman731) [inline docs/cluster-providers/generic-cluster-provider.md:1]: consider just calling this `ClusterProvider`?

**Review by kcoleman731 — COMMENTED** [review]
Looks great. Minor comments.

File: docs/cluster-providers/generic-cluster-provider.md (hunk 1 above)
Comment (shapirov103) [reply docs/cluster-providers/generic-cluster-provider.md:1]: ClusterProvider is the name of the interface that it implements. I would not want change the interface name atm, it is part of our stable SPI.

File: docs/cluster-providers/mng-cluster-provider.md [hunk 2]
```diff
@@ -21,21 +21,24 @@ new ssp.EksBlueprint(scope, { id: 'blueprint', [], [], clusterProvider });
 
 ## Configuration
 
-THe `EC2ClusterProvider` supports the following configuration options. 
+THe `MngClusterProvider` supports the following configuration options.
```
Comment (allamand) [inline docs/cluster-providers/mng-cluster-provider.md:24]: THe -> The

File: docs/addons/container-insights.md [hunk 3]
```diff
@@ -34,15 +34,16 @@ new EksBlueprint(app, 'my-stack-name', addOns, [], {
 Once the Container Insights add-on has been installed in your cluster, validate that the CloudWatch Agent and the FluentD daemons are running.
```
Comment (allamand) [inline docs/addons/container-insights.md:34]: FluentD -> fluentBit ?

File: lib/cluster-providers/generic-cluster-provider.ts [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+ */
+export interface GenericClusterProviderProps extends eks.CommonClusterOptions {
+
+    /**
+     * Whether API server is private.
+     */
+    privateCluster?: boolean,
+
+    /**
+     * Array of managed nide groups.
```
Comment (allamand) [inline lib/cluster-providers/generic-cluster-provider.ts:30]: Node -> node

File: lib/cluster-providers/generic-cluster-provider.ts [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+     * @param nodeGroup 
+     * @returns 
+     */
+    addAutoScalingGroup(cluster: eks.Cluster, nodeGroup: AutoscalingNodeGroup): autoscaling.AutoScalingGroup {
+        const machineImageType = nodeGroup.machineImageType ?? eks.MachineImageType.AMAZON_LINUX_2;
+        const instanceType = nodeGroup.instanceType ?? valueFromContext(cluster, constants.INSTANCE_TYPE_KEY, constants.DEFAULT_INSTANCE_TYPE);
+        const minSize = nodeGroup.minSize ?? valueFromContext(cluster, constants.MIN_SIZE_KEY, constants.DEFAULT_NG_MINSIZE);
+        const maxSize = nodeGroup.maxSize ?? valueFromContext(cluster, constants.MAX_SIZE_KEY, constants.DEFAULT_NG_MAXSIZE);
+        const desiredSize = nodeGroup.desiredSize ?? valueFromContext(cluster, constants.DESIRED_SIZE_KEY, minSize);
+        const updatePolicy = UpdatePolicy.rollingUpdate();
```
Comment (allamand) [inline lib/cluster-providers/generic-cluster-provider.ts:178]: Can we configure batch size for rolling update ?

**Review by allamand — APPROVED** [review]
Thanks for the hard work, that is exactly what I needed

File: lib/cluster-providers/generic-cluster-provider.ts (hunk 5 above)
Comment (shapirov103) [reply lib/cluster-providers/generic-cluster-provider.ts:178]: great idea.

Comment (shapirov103) [reply lib/cluster-providers/generic-cluster-provider.ts:178]: @allamand  This option is unfortunately not exposed through CDK at the moment, here is what they have [available](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_eks.NodegroupOptions.html). It is surely a miss. I will create an issue for that for CDK or i will move this to a lower level construct that will use direct EKS API. Thanks for bringing it up.

Comment (shapirov103) [reply lib/cluster-providers/generic-cluster-provider.ts]: addressed for ASG in the last commit, you can pass your update policy, set replace or rolling with options. 
For managed node groups, however, there is no direct support in CDK (including v2), other than falling back to lower level (CFN) constructs. I created https://github.com/aws-quickstart/ssp-amazon-eks/issues/309 to address.

**Pull request merged**